src/bootstrap.jl

```diff
@@ -89,11 +94,12 @@ function parametricbootstrap(
     # we use locks to guarantee thread-safety, but there might be better ways to do this for some RNGs
     # see https://docs.julialang.org/en/v1.3/manual/parallel-computing/#Side-effects-and-mutable-function-arguments-1
     # see https://docs.julialang.org/en/v1/stdlib/Future/index.html
-    rnglock = ReentrantLock()
+    rnglock = Threads.SpinLock()
     samp = replicate(n, use_threads=use_threads) do
-        mod = m_threads[Threads.threadid()]
-        local βsc = βsc_threads[Threads.threadid()]
-        local θsc = θsc_threads[Threads.threadid()]
+        tidx = use_threads ? Threads.threadid() : 1
```

Comment:
Good catch that the `threadid()` can be other than 1 when nested.

Reply:
The original error came from @behinger actually using this within a threaded `for` loop.